data api/src/main/java/de/tinf15b4/kino/data/ServiceImplModel.java

```diff
@@ -53,24 +52,24 @@ public E findOne(long id) {
 
         Optional<E> oe = save(e);
         if (oe.isPresent()) {
-            e = oe.get();
+            E updated = oe.get();
 
-            long newid = e.getId();
+            long newid = updated.getId();
 
             if (newid != oldid && oldid != 0) {
-                String table = e.getClass().getAnnotation(Table.class).name();
+                String table = updated.getClass().getAnnotation(Table.class).name();
 
                 // detach old entity
                 entityManager.flush();
-                entityManager.detach(e);
+                entityManager.detach(updated);
 
                 // adjust the id
-                entityManager.createNativeQuery("UPDATE " + table + " SET id = " + oldid + " WHERE id = " + newid).executeUpdate();
+                entityManager.createNativeQuery(String.format("UPDATE %s SET id = %s WHERE id = %s", table, oldid, newid)).executeUpdate();
```

Comment:
I'm inclined to believe that sonarqube had another resolution of this alleged SQL injection vulnerability in mind.

If it makes sonarqube shut up then I guess we'll keep it, but you really could have gone the extra mile here and make use of proper query parameters.